lib/open_api_spex/cast/error.ex

```diff
@@ -1,6 +1,75 @@
 defmodule OpenApiSpex.Cast.Error do
   alias OpenApiSpex.TermType
 
+  @type exclusive_max_error :: {:exclusive_max, non_neg_integer(), non_neg_integer()}
+  @type exclusive_min_error :: {:exclusive_min, non_neg_integer(), non_neg_integer()}
+  @type invalid_enum_error :: {:invalid_enum}
+  @type invalid_format_error :: {:invalid_format, any()}
+  @type invalid_schema_error :: {:invalid_schema_type}
+  @type invalid_type_error :: {:invalid_type, String.t() | atom()}
+  @type max_items_error :: {:max_items, non_neg_integer(), non_neg_integer()}
+  @type max_length_error :: {:max_length, non_neg_integer()}
+  @type max_properties_error :: {:max_properties, non_neg_integer(), non_neg_integer()}
+  @type maximum_error :: {:maximum, integer(), integer()}
+  @type min_items_error :: {:min_items, non_neg_integer(), non_neg_integer()}
+  @type min_length_error :: {:min_length, non_neg_integer()}
+  @type minimum_error :: {:minimum, integer(), integer()}
+  @type missing_field_error :: {:missing_field, String.t() | atom()}
+  @type multiple_of_error :: {:multiple_of, non_neg_integer(), non_neg_integer()}
+  @type null_value_error :: {:null_value}
+  @type unexpected_field_error :: {:unexpected_field, String.t() | atom()}
+  @type unique_items_error :: {:unique_items}
+
+  @type reason ::
+          :invalid_schema_type
+          | :exclusive_max
+          | :exclusive_min
+          | :invalid_enum
+          | :invalid_format
+          | :invalid_type
+          | :max_items
+          | :max_length
+          | :max_properties
+          | :maximum
+          | :min_items
+          | :min_length
+          | :minimum
+          | :missing_field
+          | :multiple_of
+          | :null_value
+          | :unexpected_field
+          | :unique_items
+
+  @type reasons ::
```

Comment:
Above is the type `reason`, and here is `reasons`. Without reading the definition, I would think `reasons` is a collection of `reason`. But that's not the case. This is really the reason plus details. Maybe args?